src/Makefile

```diff
@@ -494,14 +494,15 @@ ifdef COMPCXX
 	CXX=$(COMPCXX)
 endif
 
-### Sometimes gcc is really clang
+### Sometimes gcc is really clang or mingw
 ifeq ($(COMP),gcc)
 	gccversion = $(shell $(CXX) --version)
 	gccisclang = $(findstring clang,$(gccversion))
 	ifneq ($(gccisclang),)
 		profile_make = clang-profile-make
 		profile_use = clang-profile-use
 	endif
+	gccismingw = $(findstring MSYS2, $(gccversion))
```

Comment:
There are several SW distributions on Windows (msys2, cygwin, tdm-gcc etc.) with the mingw compilers. The mingw can be installed also on Linux.

Reply:
I see. Do you if there is a canonical way to identify if we are compiling in a mingw environment?

I don't have Windows computer handy, so this patch is fairly narrowly tailored to the environments tested in the CI system.

Reply:
Some googling brought me to the -dumpmachine flag, which should work for this purpose. I have added a patch to this branch.

Reply:
@ppigazzini the update patch works fine in CI and in my usual set of crosses. Does it look correct to you?

Reply:
If my fast tests with msys2 mingw32 (32bit) are correct, with master:
- `make -j build ARCH=x86-32 COMP=mingw` : doesn't use LTO
- `make -j build ARCH=x86-32 COMP=gcc` : uses LTO
This patch doesn't use LTO in both cases.
On my setup the 32 bit LTO build works fine, though, so I will prefer to have a LTO build in both cases simply dropping:
https://github.com/official-stockfish/Stockfish/blob/e1c79b5e1ed73077e502aa025f5d42a7a62f9b0d/src/Makefile#L702

A note: I will prefer to move at a higher level possible the smart `gccismingw` check in order to set `COMP=mingw` and have only one path for the build with mingw. Same for the `gccisclang` if possible.

Reply:
well, I guess the x86-32 build problems are exposed with https://github.com/official-stockfish/Stockfish/pull/3949 ?

Reply:
https://github.com/official-stockfish/Stockfish/pull/3949#issuecomment-1053702961 :
- the 32 bit LTO build always worked fine in CI (same on my pc)
- the CI should have `comp=mingw` and not `comp=gcc`

By the way we should check if there is still a difference in Makefile paths `comp=mingw` and `comp=gcc`

Reply:
Yes, LTO was what was causing that build failure. Since it was supposed to be disabled in the mingw 32-bit case already, I decided to try disabling it consistently. This was discussed somewhat at the time: https://github.com/official-stockfish/Stockfish/pull/3070#issuecomment-682148824

Obviously, what the compiler binary is named shouldn't affect which options are used, but if it is preferd that mingw 32-bit always uses LTO, rather than never using LTO, we can make that change instead, and I can try and find a different solution for https://github.com/official-stockfish/Stockfish/pull/3949

Reply:
IMO the real error is that we are not using `comp=mingw` in CI file.

Reply:
@Chaneybenjamini could you try the suggestion to change to `comp=mingw` in the CI file in the other PR?

Reply:
@vondele https://github.com/official-stockfish/Stockfish/runs/5366521608?check_suite_focus=true

Reply:
I think the install part fails, and that should continue to use gcc, only the SF build should be changed to use mingw.

Reply:
It looks like comp is used to determine the binary name, which is probably why it was set to gcc in the first place.

Reply:
@vondele I'm not sure how to do that. The tests don't seem to be setting comp explicitly. The broader issue might be related to how the Makefile sets comp when it isn't set explicitly.

Reply:
Hmm, It looks like the issue might be coming from here: https://github.com/official-stockfish/Stockfish/blob/master/.github/workflows/stockfish.yml#L15

Reply:
@Chaneybenjamini Maybe that's the way to do what @ppigazzini suggested? 
https://github.com/official-stockfish/Stockfish/pull/3953

Reply:
can you try to base your generalized feature transformer patch on that one, and see if that makes the CI pass cleanly?

Reply:
it is unfortunate your PR ran into this issue... rather tricky and rather messy.

Reply:
@vondele https://github.com/official-stockfish/Stockfish/pull/3955

Reply:
I think there is likely a bug in mingw. I did a bit more experimenting offline and its behavior is completely inexplicable.